File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/specs/GivenASetOfLibraries/AndSomeSyncOnlyOnLocalConnections/WhenGettingALibraryThatSyncsOnLocalConnection.java
package com.lasthopesoftware.bluewater.client.stored.library.specs.GivenASetOfLibraries.AndSomeSyncOnlyOnLocalConnections;

import com.lasthopesoftware.bluewater.client.library.access.specs.FakeLibraryProvider;
import com.lasthopesoftware.bluewater.client.library.repository.Library;
import com.lasthopesoftware.bluewater.client.library.repository.LibraryId;
import com.lasthopesoftware.bluewater.client.stored.library.SyncLibraryProvider;
import com.lasthopesoftware.bluewater.shared.promises.extensions.specs.FuturePromise;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;
import java.util.concurrent.ExecutionException;

import static org.assertj.core.api.Assertions.assertThat;

public class WhenGettingALibraryThatSyncsOnLocalConnection {

	private static Library library;

	@BeforeClass
	public static void context() throws ExecutionException, InterruptedException {
		final SyncLibraryProvider syncLibraryProvider = new SyncLibraryProvider(
			new FakeLibraryProvider(Arrays.asList(
				new Library().setId(3),
				new Library().setId(4),
				new Library().setId(8).setIsSyncLocalConnectionsOnly(true),
				new Library().setId(1)
			)));

		library = new FuturePromise<>(syncLibraryProvider.getLibrary(new LibraryId(8))).get();
	}

	@Test
	public void thenTheLibraryIsLocalOnly() {
		assertThat(library.isLocalOnly()).isTrue();
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/stored/library/GivenASetOfLibraries/AndSomeSyncOnlyOnLocalConnections/WhenGettingALibraryThatSyncsOnLocalConnection.kt
package com.lasthopesoftware.bluewater.client.stored.library.GivenASetOfLibraries.AndSomeSyncOnlyOnLocalConnections

import com.lasthopesoftware.bluewater.client.browsing.library.access.FakeLibraryProvider
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.stored.library.SyncLibraryProvider
import com.lasthopesoftware.bluewater.shared.promises.extensions.ExpiringFuturePromise
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class WhenGettingALibraryThatSyncsOnLocalConnection {
	private val library by lazy {
		val syncLibraryProvider = SyncLibraryProvider(
			FakeLibraryProvider(
				Library().setId(3),
				Library().setId(4),
				Library().setId(8).setIsSyncLocalConnectionsOnly(true),
				Library().setId(1),
				Library().setId(13).setIsSyncLocalConnectionsOnly(true)
			)
		)
		ExpiringFuturePromise(syncLibraryProvider.promiseLibrary(LibraryId(8))).get()
	}

	@Test
	fun `then the library is local only`() {
		assertThat(library!!.isLocalOnly).isTrue
	}
}
